### ISSUE TITLE
[Edit] - "Evaluation"-Chapter

### ISSUE THREAD

**Issue by henningsway** [Tazinho/Advanced-R-Solutions#137]
## [Evaluation](https://advanced-r-solutions.rbind.io/evaluation.html)

@Tazinho 
- [x] Sync subsection names
- [x] Ensure that exercise numbers go 1,2,3,.. in each subsection.
- [x] Ensure linebreaks between Q & As
- [x] Sync the text and the order of all exercises in the chapter between Adv R and Adv R Solutions.

Read through all exercises. Edit them if possible and mark their correct status (green = "think this solution is correct", orange = "this solution is definitely not correct or sth. is missing". If sth. is missing, try to includede a todo). Also document open points within the following checklist:

### [Evaluation basics](https://advanced-r-solutions.rbind.io/quasiquotation.html#evaluation-basics)
- [x] Carefully read the documentation for source(). What environment does it use by...
- [x] Predict the results of the following lines of code:...
* need to adjust the exercise text as well as the answer (`expr()` instead of `quote()`)
* I think the pattern needs to be described a bit more detailled and compact at the same time. I tried to break it down to get a more clear overview. Unfortunately without good result till now.
- [x] Fill in the function bodies below to re-implement get() using sym()
- [x] Modify source2() so it returns the result of every expression, not...
* need to read more detailled. Why are there errors returned? Also a bit hard to follow. Need to check the context regarding the textbook.
- [ ] We can make base::local() slightly easier to understand by spreading...
* Is started and I left a comment (directly as part of the answer). Maybe you or Hadley can help here.
### [Quosures](https://advanced-r-solutions.rbind.io/quasiquotation.html#quosures)
- [x] Predict what each of the following quosures will return if evaluated....
- [x] Write an enenv() function that captures the environment associated w...
* looks not bad, but not sure if the 2nd test makes sense, i.e. to capture the calling environment of `enenv()` (not sure if the implementation might be even wrong, and should instead capture the environment of the argument...)
* however, also if correct, the formatting of the code comments needs to be a bit more structured.

### [Data masks](https://advanced-r-solutions.rbind.io/quasiquotation.html#data-masks)
- [ ] Why did I use a for loop in transform2() instead of map()? Consider ....
* text could probably still be a "little" bit enhanced.
- [ ] Here’s an alternative implementation of subset2():...
* I edited as good as possible. However, I didn't understand it completely. Could you add what is the data mask concretely in the case where the error occures. I.e. add it after the line "... This means, that the subsetting is also evaluated in the context of the data mask." (I would think that the data mask is df, but data and df are the same, so there should be no difference...obv I am getting confused here :D). And afterwards edit as much as you think is needed.
- [x] The following function implements the basics of dplyr::arrange()...
* needs to be reviewed in more detail. Possibly text/textstructure can also be a bit enhanced

### [Using tidy evaluation](https://advanced-r-solutions.rbind.io/quasiquotation.html#using-tidy-evaluation)
- [ ] I’ve included an alternative implementation of threshold_var() below....
* Hm,... I would have written almost the same. But it really seems that sth. missing. Maybe there is a hint in the passage within the textbook? (I didn't make it. Maybe you are more lucky?)
* Not sure if it's good to add the suffix ("2") in the exercise version of `threshold_var()`. Don't have a better idea, but would be nicer if we added sth. to the older version of `threshold_var` (just a thought, probably not that important).

### [Base evaluation](https://advanced-r-solutions.rbind.io/quasiquotation.html#base-evaluation)
- [ ] Why does this function fail?
* Sounds good :) Maybe you can add a comment in the end, which alternative is preferable?
- [x] When model building, typically the response and data are relatively constant...
- [ ] Another way to write resample_lm() would be to include the resample...
* I didn't touch this one, as I think I don't understand the exact details from the last sections enough. Would be great if you could just have a look at this one again.

- [x] All open points/exercises should now be reported explicitly under each bullet point above (maybe henningsway can solve one or the other, has a good idea for a new approach or has a more specific question to raise in an issue to Hadley. 

- [x] Assign Henningsway

@henningsway 
- [ ] Read & edit all exercises.
- [ ] Report any new points while updating the list above. 
- [ ] Run a final spellcheck

@Tazinho and @henningsway 
- [ ] Try to resolve any of the reported issues above.

**Comment by Tazinho** [comment]
@henningsway finished as good as I could. Hope you can resolve some of the outstanding issues.
